src/platforms/web/compiler/directives/model.js

```diff
@@ -93,7 +93,7 @@ function genCheckboxModel (
     'if(Array.isArray($$a)){' +
       `var $$v=${number ? '_n(' + valueBinding + ')' : valueBinding},` +
           '$$i=_i($$a,$$v);' +
-      `if($$el.checked){$$i<0&&(${value}=$$a.concat($$v))}` +
+      `if($$el.checked){$$i<0&&(${value}=$$a.concat(Array.isArray($$v)?[$$v]:$$v))}` +
```

Comment:
I think using `[$$v]` is a safer way of doing it. No need for ternary imo
You can add a comment if you want